src/speculators/models/eagle.py

```diff
@@ -532,7 +529,45 @@ def _create_transformer_layer(self) -> nn.Module:
         return layer
 
     def _layernorm_class(self) -> type[nn.Module]:
-        return LlamaRMSNorm
+        return self._architecture_classes[1]
 
     def _transformer_layer_class(self) -> type[nn.Module]:
-        return LlamaDecoderLayer
+        return self._architecture_classes[0]
+
+    def _import_model_classes(self) -> tuple[type, type]:
+        """
+        Get the decoder layer class and layer normalization class for a given config
+        and decoder layer name. Uses the config qualified name to find the corresponding
+        modeling module to import the decoder layer and normalization classes from.
+        """
+        # e.g. LlamaConfig + LlamaDecoderLayer
+        # -> get qualname: transformers.models.llama.configuration_llama.LlamaConfig
+        # -> expected path: transformers.models.llama.modeling_llama.LlamaDecoderLayer
+        # -> expected path: transformers.models.llama.modeling_llama.LlamaRMSNorm
+
+        module_path = self.config.transformer_layer_config.__class__.__module__
```

Comment:
I'm worried about how delicate this is since it's an assumption that is not enforced by the Transformers repo. A pathway that better matches Transformers' internal logic is determining the module that contains the model class by utilizing the MODEL_FOR_CAUSAL_LM_MAPPING from transformers and looking up by either the config class or the architecture contained in the architectures property.

Reply:
Yeah, this was not my favorite implementation but in practice it does seem to hold. I ran a test script that found every instance of ".*DecoderLayer" in the transformers model directory and the matching config class, and then checked if the classes could load correctly and it worked every time. Also a lot of the `modeling_*` modules are auto-generated from a `modular_*` module in the same folder. So in practice this pattern is pretty consistent. 

That being said, using `MODEL_FOR_CAUSAL_LM_MAPPING` to get to modeling module should be a bit more robust, so I will try using that instead.